components/webext-storage/src/sync/incoming.rs

```diff
@@ -236,12 +262,17 @@ pub fn plan_incoming(s: IncomingState) -> IncomingAction {
                     // Treat this as a delete of every key that we
                     // knew was present. Unfortunately, we don't know
                     // any keys that were present, so we delete no keys.
-                    IncomingAction::Merge { data: local_data }
+                    IncomingAction::Merge {
+                        data: local_data,
+                        changes: StorageChanges::new(),
+                    }
                 }
                 (DataState::Deleted, DataState::Deleted, _) => {
                     // We agree with the remote (regardless of what we
                     // have mirrored).
-                    IncomingAction::DeleteLocally
+                    IncomingAction::DeleteLocally {
```

Comment:
Seems like this should be `DataState::Same` for clarity (although the end result should be the same)

Reply:
Nice catch! Changed.